src/tests/stats/test_corr.fypp

```diff
@@ -0,0 +1,280 @@
+#:include "common.fypp"
+#:set IR_KINDS_TYPES = INT_KINDS_TYPES + REAL_KINDS_TYPES
+
+module test_stats_corr
+    use stdlib_test, only : new_unittest, unittest_type, error_type, check
+    use stdlib_stats, only: corr
+    use stdlib_kinds, only : int8, int16, int32, int64, sp, dp, qp
+    use,intrinsic :: ieee_arithmetic, only : ieee_is_nan
+    implicit none
+    private
+  
+    public :: collect_stats_corr
+
+    #:for k1, t1 in REAL_KINDS_TYPES
+    ${t1}$, parameter :: ${k1}$tol = 1000 * epsilon(1._${k1}$)
+    #:endfor
+
+    #:for k1, t1 in IR_KINDS_TYPES
+    ${t1}$, parameter :: d1_${k1}$(5) = [${t1}$ :: 1, 2, 3, 4, 5]
+    ${t1}$, parameter :: d2_${k1}$(4, 3) =&
+                                   reshape([${t1}$ :: 1, 3, 5, 22,&
+                                   3, 4, 6, 20,&
+                                   15, 14, 13, 12], [4, 3])
+    #:endfor
+
+    #:for k1, t1 in CMPLX_KINDS_TYPES
+    complex(${k1}$), parameter :: c1_${k1}$(5) =&
+                            [ cmplx(0.57706_${k1}$, 0.00000_${k1}$, kind = ${k1}$),&
+                            cmplx(0.00000_${k1}$, 1.44065_${k1}$, kind = ${k1}$),&
+                            cmplx(1.26401_${k1}$, 0.00000_${k1}$, kind = ${k1}$),&
+                            cmplx(0.00000_${k1}$, 0.88833_${k1}$, kind = ${k1}$),&
+                            cmplx(1.14352_${k1}$, 0.00000_${k1}$, kind = ${k1}$)]
+    complex(${k1}$), parameter :: c2_${k1}$(2,3) &
+                            = reshape([ cmplx(1._${k1}$, 0._${k1}$, kind = ${k1}$),&
+                            cmplx(0._${k1}$, 2._${k1}$, kind = ${k1}$),&
+                            cmplx(3._${k1}$, 0._${k1}$, kind = ${k1}$),&
+                            cmplx(0._${k1}$, 4._${k1}$, kind = ${k1}$),&
+                            cmplx(5._${k1}$, 0._${k1}$, kind = ${k1}$),&
+                            cmplx(0._${k1}$, 6._${k1}$, kind = ${k1}$)], [2, 3])
+    #:endfor
+
+contains
+
+    !> Collect all exported unit tests
+    subroutine collect_stats_corr(testsuite)
+        !> Collection of tests
+        type(unittest_type), allocatable, intent(out) :: testsuite(:)
+
+        testsuite = [ &
+            new_unittest("test_${k1}$_1dim", test_${k1}$_1dim) &
```

Comment:
Is this intended? `k1` would be undefined here, wouldn't it?

Reply:
No. you are right. It should be replaced by `sp`.